remove_library.php

```diff
@@ -0,0 +1,94 @@
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Remove h5p library.
+ *
+ * @package    mod_hvp
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2022 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require(__DIR__ . '/../../config.php');
+global $CFG, $DB, $PAGE, $OUTPUT;
+require_once($CFG->libdir . '/adminlib.php');
+require_once($CFG->libdir . '/modinfolib.php');
+require_once($CFG->dirroot . '/course/lib.php');
+require_once($CFG->dirroot . '/mod/hvp/lib.php');
+
+require_admin();
+
+$options = [
+    'help' => optional_param('help', false, PARAM_BOOL),
+    'library-id' => optional_param('library-id', '', PARAM_ALPHANUMEXT),
+    'library-title' => optional_param('library-title', '', PARAM_ALPHANUMEXT),
+    'major-version' => optional_param('major-version', '', PARAM_ALPHANUMEXT),
+    'minor-version' => optional_param('minor-version', '', PARAM_ALPHANUMEXT),
+    'patch-version' => optional_param('patch-version', '', PARAM_ALPHANUMEXT),
+    'run' => optional_param('run', false, PARAM_BOOL),
+];
+
+$output = '';
+
+if ((empty($options['library-id'])
+            && empty($options['library-title'])
+            && empty($options['major-version'])
+            && empty($options['minor-version'])
+            && empty($options['patch-version']))
+        || $options['help']) {
+    $output = get_string('query:help', 'hvp');
+} else {
+    $framework = \mod_hvp\framework::instance('interface');
+
+    // Get the installed library objects. TODO: Allow multiple ids.
+    $library = \mod_hvp\helper::get_library($options);
+
+    // 1. Check if there is any dependency on the lib about to be deleted and stop process with message if one is found.
+    $dependantlibraries = \mod_hvp\helper::get_dependant_libraries($library);
+    if (!empty($dependantlibraries)) {
+        throw new moodle_exception('error:librarydependency', 'hvp', '', implode(',', $dependantlibraries));
+    } else {
+        $output .= html_writer::tag('p', "Library '{$library->title} {$library->major_version}.{$library->minor_version}.{$library->patch_version}' can be deleted.");
+    }
+
+    // 2. Search for all activities that are using the lib and delete them.
+    try {
+        $countactivities = \mod_hvp\helper::remove_dependant_activities($library, !$options['run']);
```

Comment:
Same here, lets expose the activity count.

Reply:
Message here: https://github.com/catalyst/moodle-mod_hvp/pull/20/files/85fd8a97890d2a881f6f020b9b78e82afffe10ae#diff-49508de1afafd6780bd453b3dbfc9898b0fafc1ff6fbbd45dfb47b2ef53dc8e4R75